src/Admin/UserManager.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Class UserManager.
+ *
+ * @since 1.6.0
+ *
+ * @package AMP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Service;
+
+/**
+ * Class UserManager
+ *
+ * @since 1.6.0
+ */
+final class UserManager implements Service {
+
+	/**
+	 * Custom user capability allowing dev tools
+	 *
+	 * @var string
+	 */
+	const VALIDATE_CAP = 'amp_validate';
+
+	/**
+	 * User meta key enabling or disabling developer tools.
+	 *
+	 * @var string
+	 */
+	const USER_OPTION_DEVELOPER_TOOLS = 'amp_dev_tools_enabled';
+
+	/**
+	 * Sets up hooks.
+	 */
+	public function register() {
+		add_filter( 'amp_setup_wizard_data', [ __CLASS__, 'inject_setup_wizard_data' ] );
+		add_action( 'rest_api_init', [ __CLASS__, 'register_user_meta' ] );
+		add_filter( 'get_user_metadata', [ __CLASS__, 'get_default_enable_developer_tools_setting' ], 10, 3 );
+		add_filter( 'update_user_metadata', [ __CLASS__, 'update_enable_developer_tools_permission_check' ], 10, 4 );
+	}
+
+	/**
+	 * Registers user meta related to validation management.
+	 *
+	 * @since 1.6.0
+	 */
+	public static function register_user_meta() {
+		register_meta(
+			'user',
+			self::USER_OPTION_DEVELOPER_TOOLS,
+			[
+				'show_in_rest' => true,
+				'single'       => true,
+				'type'         => 'boolean',
+			]
+		);
```

Comment:
After seeing how this has to be implemented with straight user meta, I think we should actually implement what @pierlon did for `amp_enabled` for posts. Instead of exposing the postmeta in REST, he instead added a new rest field on the post entity. This will make it way simpler without having to do a dance with `get_user_metadata` and `update_user_metadata`. Instead, we'll be able to easily do `register_rest_field( 'user', self::USER_OPTION_DEVELOPER_TOOLS, $args )`. 

Then the `get_callback` can implement the default behavior like:

```php
if ( ! AMP_Validation_Manager::has_cap() ) { // @todo This is wrong, needs to user_has_cap( 'amp_validate', $user_id )
    return false;
}
return (bool) get_user_meta( $user_id, self::USER_OPTION_DEVELOPER_TOOLS, true );
```

The `update_callback` would need to return a `WP_Error` if the user cannot `amp_validate`.

Reply:
In the case of a user who cannot validate in the first place, the `get_callback` can also return `null`. That would differentiate the user from one who `has_cap` but just has Dev Tools turned off. In other words, when a user has Dev Tools turned off, we still intend to allow them to access the Dev Tools. It's just we won't proactively display the UI.

Then again, maybe that's overloading the field. We should instead look at the `capabilities` object being sent back with the user to see if it includes `amp_validate`.

Reply:
Good idea. I'm going to give it a shot with `register_rest_field`.

Reply:
`register_rest_field` is working out fine. I think it's going to work best if we do it this way:

1. The `get_callback` will return true or false if the `amp_dev_tools_enabled` has previously been set, but it will return `null` if it has not been set. Then when the user arrives at the technical question in the wizard, nothing will be pre-selected if the value is `null`. If the value is true/false, the corresponding answer will be pre-selected.
2. In the `update_callback`, if the new value is true we check that the user being updated (not necessarily the current user) can manage options or validate amp. If they can't, we return an error. Of course this will probably never happen because you need to be able to manage options to get to the setup wizard.
3. Beyond that, I think the permissions around accessing the UI around validation should happen elsewhere, e.g., with a permissions check around the UI handler. So in theory a user without permission to use dev tools could have `amp_dev_tools_enabled` set to anything, and it wouldn't matter because they would be blocked by a capability check.

Reply:
I think that sounds fine.